File: 0056-merge-intervals/0056-merge-intervals.py
class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        result = []
        sortedInt = sorted(intervals)
        currentInterval = sortedInt[0]
        for interval in sortedInt[1:]:
            if currentInterval[1] >= interval[0]:
                currentInterval[1] = max(interval[1], currentInterval[1])
            else:
                result.append(currentInterval)  
                currentInterval = interval
        result.append(currentInterval)
        return result          

File: 0056-merge-intervals/0056-merge-intervals.cpp
#include <algorithm>
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        std::sort(intervals.begin(), intervals.end()); 
        vector<vector<int>> res;
        res.push_back(intervals[0]);
        for(vector<int> interval : intervals)
        {
            vector<int> arr = res.back();
            if(arr[1] >= interval[0] && arr[1] <= interval[1])
            {
                res.back() = {arr[0], interval[1]};
            }
            else if(arr[0] <= interval[0] && arr[1] >= interval[1])
            {
                res.back() = arr;
            }
            else
            {
                res.push_back(interval);
            }
        }
        return res;
    }
};